### PR TITLE
Decide score should read weights - BEHROUZ -#101

### LINKED ISSUE #101: `decideScore` should read weighting values from config
## What?
The `decideScore` function should read any weighting values from the DB config table

## Why?
We have made is possible to edit these weighting values via a UI and we want to make sure the score algorithm is using any new value set

### PR DESCRIPTION
**Pull request by bkarimii**
## Description
decideScore function updated to allow customizable weights for messages, reactions, and reactions received. Default weights are used if no custom values are provided. The function handles edge cases, such as missing or non-numeric inputs. Test cases have been updated to match the new version of the function.

## Related to

Fixes #101

### CODE REVIEW

**Comment by bkarimii** [comment]
This PR should be merged before  #92

File: api/functions/decideScore.js [hunk 1]
```diff
@@ -1,23 +1,54 @@
...
+	if (!messages) {
+		return reactionWeight * reactions;
+	}
 
-	if (!reactions) return messages * 3;
+	if (!reactions) {
+		return (
+			messageWeight * messages + reactionsReceivedWeight * reactionsReceived
+		);
+	}
```
Comment (jordan-choi) [inline api/functions/decideScore.js]: Consider simplifying the logic structure as follows for better readability:
```suggestion
if (messages <= 0 && reactions <= 0) return 0;

if (!messages) return reactionWeight * reactions;

return (
  messageWeight * messages +
  reactionWeight * reactions +
  reactionsReceivedWeight * reactionsReceived
);
```

File: api/functions/decideScore.test.js [hunk 2]
```diff
@@ -6,28 +6,36 @@ describe("decideScore", () => {
...
-	it("should handle when there are no messages", () => {
-		const score = decideScore({ messages: 0, reactions: 3 });
-		expect(score).toBe(3);
+	it("should return 30 when there are 10 messages and 10 reactions and reactionsReceived = 5", () => {
+		const score = decideScore({
+			messages: 10,
+			reactions: 10,
+			reactionsReceived: 5,
+		});
+		expect(score).toBe(45);
```
Comment (jordan-choi) [inline api/functions/decideScore.test.js:25]: I noticed a discrepancy between the test comment description and the expected value in the assertion.

File: api/functions/decideScore.test.js [hunk 3]
```diff
@@ -6,28 +6,36 @@ describe("decideScore", () => {
...
 
-	it("should handle when there are no messages and no reactions", () => {
-		const score = decideScore({ messages: 0, reactions: 0 });
+	it("should return score based on reactionsReceived when no messages and no reactions", () => {
+		const score = decideScore({
+			messages: 0,
+			reactions: 0,
+			reactionsReceived: 0,
+		});
 		expect(score).toBe(0);
```
Comment (jordan-choi) [inline api/functions/decideScore.test.js]: For a more thorough test, we might want to use a case with `messages: 0, reactions: 0, reactionsReceived: Math.random() * 100` (any random positive number) to verify the behavior with varying inputs.

**Review by jordan-choi — COMMENTED** [review]
Thank you for your work on this. Hope these minor adjustments help improve both clarity and robustness of the implementation ☺️.

File: api/functions/decideScore.test.js (hunk 3 above)
Comment (bkarimii) [reply api/functions/decideScore.test.js]: Good idea. thanks.

File: api/functions/decideScore.test.js (hunk 2 above)
Comment (bkarimii) [reply api/functions/decideScore.test.js:25]: thank you for noticing it. fixed now.

**Review by jordan-choi — APPROVED** [review]
